Packs/SailPointIdentityNow/Integrations/SailPointIdentityNowEventCollector/SailPointIdentityNowEventCollector.py

```diff
@@ -0,0 +1,375 @@
+            elif created:
+                event['_time'] = created.strftime(DATE_FORMAT)
+                event["_ENTRY_STATUS"] = "new"
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """
+    main function, parses params and runs command functions
+    """
+
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+    client_id = params.get('credentials', {}).get('identifier')
+    client_secret = params.get('credentials', {}).get('password')
+    base_url = params['url']
+    verify_certificate = not params.get('insecure', False)
+    proxy = params.get('proxy', False)
+    limit = arg_to_number(params.get('limit')) or 50000
+
+    demisto.debug(f'Command being called is {command}')
+    try:
+        client = Client(
+            client_id=client_id,
+            client_secret=client_secret,
+            base_url=base_url,
+            verify=verify_certificate,
+            proxy=proxy)
+
+        if command == 'test-module':
+            result = test_module(client)
+            return_results(result)
+
+        elif command == 'identitynow-get-events':
+            limit = arg_to_number(args.get('limit', 50))
+            should_push_events = argToBoolean(args.get('should_push_events', False))
+            time_to_start = arg_to_datetime(args.get('from_date'))
+            formatted_time_to_start = time_to_start.strftime(DATE_FORMAT) if time_to_start else DEFAULT_NOW
+            id_to_start = args.get('from_id')
+            if not (id_to_start or time_to_start):
+                raise DemistoException("Either from_id or from_date must be provided.")
+            if id_to_start and time_to_start:
+                raise DemistoException("Both from_id and from_date cannot be provided.")
+            events, results = get_events(client, from_date=formatted_time_to_start,
+                                         from_id=id_to_start, limit=limit)   # type:ignore
+            return_results(results)
+            if should_push_events:
+                add_time_and_status_to_events(events)
+                send_events_to_xsiam(
+                    events,
+                    vendor=VENDOR,
+                    product=PRODUCT
+                )
+
+        elif command == 'fetch-events':
+
+            last_run = demisto.getLastRun()
+            next_run, events = fetch_events(
+                client=client,
+                limit=limit,
+                last_run=last_run,
+            )
+
+            add_time_and_status_to_events(events)
+            demisto.debug(f'Sending {len(events)} events to Xsiam.')
+            send_events_to_xsiam(
+                events,
+                vendor=VENDOR,
+                product=PRODUCT
+            )
+            demisto.setLastRun(next_run)
+            demisto.debug(f'Next run is set to: {next_run}.')
+
+    # Log exceptions and return errors
+    except Exception as e:
+        return_error(f'Failed to execute {command} command.\nError:\n{str(e)}')
```

Comment:
Lets make this the else statement and make this exception handled as in all other integrations.